lib/dry/validation/rule/attr.rb

```diff
@@ -0,0 +1,21 @@
+module Dry
+  module Validation
+    class Rule::Attr < Rule
+      def self.new(name, predicate)
+        super(name, predicate.curry(name))
+      end
+
+      def type
+        :attr
+      end
+
+      def evaluate_input(input)
+        input.public_send(name)
+      end
+
+      def call(input)
+        Validation::Rule::Result::EvaluatedValue.new(input, predicate.(input), self)
```

Comment:
How would that work? I remember we discussed this but now I can't find it for some reason :(